control.lua

```diff
@@ -521,6 +527,44 @@ local function init()
   init_events()
 end
 
+local function config_changed(changed_data)
+  local mod_changes = changed_data.mod_changes
+  local old_version
+  if mod_changes and mod_changes["cargo-ships"] and mod_changes["cargo-ships"]["old_version"] then
+    old_version = mod_changes["cargo-ships"]["old_version"]
+  else
+    return
+  end
+
+  log("Coming from old version: " .. old_version)
+  old_version = util.split(old_version, ".")
+  for i=1, #old_version do
+    old_version[i] = tonumber(old_version[i])
+  end
+
+  if old_version[1] == 0 then
+    if old_version[2] < 2 then
+      -- Pre 0.2.0
+
+      -- Teleport power entities to center of oil rigs since previously they didn't have
+      -- placeable-off-grid flag so weren't centered
+      if global.or_generators ~= nil then
+        for i, generator in pairs(global.or_generators) do
+          if generator.valid then
+            local surface = generator.surface
+            local oil_rig = surface.find_entity("oil_rig", generator.position)
+            local position = oil_rig.position
+            generator.teleport(position)
+
+            -- Add new power entity
+            surface.create_entity{name = "or_power_electric", position = position, force = oil_rig.force}
```

Comment:
Since the only commands you are running here are to teleport one entity and create another, I think a Lua migration would work just as well.

Reply:
I'll switch to that later today